### ISSUE TITLE
[AIR] Framework specific Checkpoint API consistency

### ISSUE THREAD

**Issue by amogkam** [ray-project/ray#28412]
We should make sure all of our framework-specific Checkpoints follow similar API design patterns.

- For creating checkpoints via different sources (for example, from a model object, from a path, etc.), these should be expressed as different methods and not overloaded onto the same method.
- All framework specific checkpoint classes should expose `def get_model(self, *args, **kwargs)` and `def get_preprocessor(self) -> Optional[Preprocessors]` methods. These can be defined in a private abstract base class that all framework specific checkpoint can inherit from.
- All predictors should interact with framework specific checkpoints through `get_model` and `get_preprocessor` APIs.
- The framework-specific Predictors' `from_checkpoint` method should accept `**get_model_kwargs` so users can specify kwargs that should be passed through to `checkpoint.get_model(...)`

cc @xwjiang2010

**Comment by heyitsmui** [comment]
@xwjiang2010 @krfricke im happy to take a look at this if this hasn't been grabbed!

**Comment by richardliaw** [comment]
yeah please do so! and keep us updated. opening a draft PR would be the fastest way to collaborate here!

**Comment by amogkam** [comment]
hey @heyitsmui, let me know if you are still planning to work on this :).

**Comment by heyitsmui** [comment]
hey @xwjiang2010 i believe you mentioned you already have a draft PR for this? if not, i can get started on this :)

**Comment by xwjiang2010** [comment]
Ah the one I have is this: https://github.com/ray-project/ray/pull/28474
It's currently being blocked by another PR, which should be landed soon. I dont have other parts of this task planned yet.

**Comment by amogkam** [comment]
Thanks @heyitsmui! Let me know if you have any questions!

**Comment by heyitsmui** [comment]
Working on this now, expect to have a draft PR out for review later today

**Comment by richardliaw** [comment]
awesome!

**Comment by richardliaw** [comment]
@heyitsmui did you create a pr for this?

**Comment by richardliaw** [comment]
bump @heyitsmui ?

**Comment by heyitsmui** [comment]
@richardliaw thanks for checking in, let me clean up my change, will have a PR out for review by tomorrow morning

**Comment by richardliaw** [comment]
^^ @heyitsmui :)
